### PR TITLE
RHEL 8 host test fixes

### PR DESCRIPTION
**Pull request by jelly**

### CODE REVIEW

File: test/image-prepare [hunk 1]
```diff
@@ -131,7 +131,7 @@ def build_install_rhel8(dist_tar, image, verbose):
             # build rpms in mock
             "--run-command",
             """su builder -c 'mock --no-clean --no-cleanup-after --disablerepo=* --offline --resultdir /var/tmp/build """
-            f"""--nocheck {"--verbose" if verbose else ""} --rebuild /var/tmp/build/SRPMS/*.src.rpm' """,
+            f"""{"--nocheck" if quick else ""} {"--verbose" if verbose else ""} --rebuild /var/tmp/build/SRPMS/*.src.rpm' """,
```
Comment (martinpitt) [inline test/image-prepare:134]: Thanks. Sorry for that blunder!

File: src/ws/test-channelresponse.c [hunk 2]
```diff
@@ -541,6 +541,15 @@ test_resource_failure (TestResourceCase *tc,
   g_assert (cockpit_pipe_get_pid (tc->pipe, &pid));
   g_assert_cmpint (pid, >, 0);
   int pid_fd = syscall(SYS_pidfd_open, pid, 0);
+  if (pid_fd < 0)
+    {
+      if (errno == ENOSYS)
+	g_test_skip ("no pidfd_open support, skipping");
+      else
+        g_test_skip (strcat("pidfd_open call failed: ", g_strerror (errno)));
```
Comment (martinpitt) [inline src/ws/test-channelresponse.c:549]: We shouldn't silently skip the test if that fails for any other reason, as then we have screwed up the code. Also, the strcat() here is both invalid (you can't append to a const string), and if it *would* copy, it'd be a memory leak. To fix all of these, use this:

    g_error ("pidfd_open call failed: %m");

**Review by martinpitt — CHANGES_REQUESTED** [review]

File: test/image-prepare (hunk 1 above)
Comment (jelly) [reply test/image-prepare:134]: No worries!

File: src/ws/test-channelresponse.c (hunk 2 above)
Comment (jelly) [reply src/ws/test-channelresponse.c]: Ah yes, I was not 100% certain how we would like to handle it. `g_error` makes sense.

File: src/ws/test-channelresponse.c [hunk 3]
```diff
@@ -541,6 +541,15 @@ test_resource_failure (TestResourceCase *tc,
...
   g_assert_cmpint (pid, >, 0);
   int pid_fd = syscall(SYS_pidfd_open, pid, 0);
+  if (pid_fd < 0)
+    {
+      if (errno == ENOSYS)
+	g_test_skip ("no pidfd_open support, skipping");
+      else
+        g_error ("pidfd_open call failed: %m");
+
+      return;
```
Comment (martinpitt) [inline src/ws/test-channelresponse.c:551]: This is slightly "meh"/confusing, as g_error() aborts. It'd be a bit cleaner to move this into the "then" branch. But it doesn't matter in practice (not sure how strongly @allisonkarlitskaya feels about that). Thanks!

**Review by martinpitt — APPROVED** [review]

File: src/ws/test-channelresponse.c (hunk 3 above)
Comment (jelly) [reply src/ws/test-channelresponse.c:551]: Ah, I guess it's only logical when you don't know the glib api

File: src/ws/test-channelresponse.c [hunk 4]
```diff
@@ -541,6 +541,15 @@ test_resource_failure (TestResourceCase *tc,
...
+    }
```
Comment (allisonkarlitskaya) [inline src/ws/test-channelresponse.c:552]: I think I'd write it

```suggestion
  if (pid_fd < 0)
    {
      if (errno != ENOSYS)
        g_error ("pidfd_open call failed: %m");
        
      g_test_skip ("no pidfd_open support, skipping");
      return;
    }
```

But I don't think it's important.  I'd be fine with yours.

File: test/image-prepare (hunk 1 above)
Comment (allisonkarlitskaya) [inline test/image-prepare:134]: Oof.  Thanks.

**Review by allisonkarlitskaya — APPROVED** [review]

**Comment by martinpitt** [comment]
Ack, let's land this then, as it's a release blocker.

**Pull request merged**